// venv/lib/environment-template-handler.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const ejs = require("ejs");
const fs = require("fs-extra");
const ef8 = {
    encoding: 'utf-8',
};
function preprocess(envContainer, predicate = s => s) {
    const result = Object.keys(envContainer.env).reduce((p, c) => {
        Object.assign(p.env, {
            [c]: predicate(envContainer.env[c]),
        });
        return p;
    }, {
        env: {},
    });
    return result;
}
class EnvironmentTemplateHandler {
    renderEnvContextTemplateFile(path, envContainer, callback) {
        if (callback) {
            fs.readFile(path, ef8, (err, templatBuffer) => {
                if (err) {
                    callback(err);
                }
                else {
                    const template = templatBuffer.toString();
                    callback(null, EnvironmentTemplateHandler.renderEnvContext(template, envContainer));
                }
            });
        }
        else {
            return fs
                .readFile(path, ef8)
                .then(template => {
                return EnvironmentTemplateHandler.renderEnvContext(template, envContainer);
            })
                .catch(e => {
                throw e;
            });
        }
    }
    renderEnvContextTemplateFileSync(path, envContainer) {
        const template = fs.readFileSync(path, ef8);
        return EnvironmentTemplateHandler.renderEnvContext(template, envContainer);
    }
    static renderEnvContext(template, envContainer) {
        const compiled = ejs.compile(template);
        const result = compiled(preprocess(envContainer));
        return result;
    }
}
exports.EnvironmentTemplateHandler = EnvironmentTemplateHandler;
