// venv/lib/venv.js
#!/usr/bin/env node
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
const fs = require("fs-extra");
const yargs = require("yargs");
const environment_template_handler_1 = require("./environment-template-handler");
const argument_parser_1 = require("./argument-parser");
yargs
    .command('process', 'replace template file with environment variables', y => {
    return y.option('t', {
        alias: 'templatePath',
        describe: 'the template file',
        nargs: 1,
        required: true,
    });
}, replaceWithTemplate)
    .example('$0 process -t foo.js', 'replace the specified file with its processed .template')
    .demandCommand(1)
    .option('h', {
    alias: 'help',
    description: 'display help message',
})
    .epilog('for more information see https://github.com/GaryB432/venv')
    .alias('v', 'version')
    .version(() => require('../package').version)
    .describe('v', 'show version information')
    .help().argv;
function replaceWithTemplate(yav) {
    return __awaiter(this, void 0, void 0, function* () {
        const opts = argument_parser_1.readArguments(yav);
        const eth = new environment_template_handler_1.EnvironmentTemplateHandler();
        eth
            .renderEnvContextTemplateFile(opts.filePath.concat('.template'), process)
            .then(data => {
            fs.writeFile(opts.filePath, data).then(() => console.log(opts.filePath));
        })
            .catch(e => {
            console.log(e.message);
        });
    });
}
